tests/Microsoft.Graph.DotnetCore.Core.Test/Helpers/ExtractSelectHelperTest.cs

```diff
@@ -0,0 +1,169 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Graph.DotnetCore.Core.Test.Helpers
+{
+    public class ExtractSelectHelperTest
+    {
+        /// <summary>
+        /// Simple base class to use as a template parameter for testing
+        /// </summary>
+        private class EventBase
+        {
+            public DateTimeOffset CreatedDateTime { get; set; }
+        }
+
+        /// <summary>
+        /// Simple class to use as a template parameter for testing
+        /// </summary>
+        private class Event : EventBase
+        {
+            public string Body { get; set; }
+
+            public string Subject { get; set; }
+
+        }
+
+        /// <summary>
+        /// Simple recursive class to use as a template parameter for testing
+        /// </summary>
+        private class User
+        {
+            public string DisplayName { get; set; }
+
+            public User Manager { get; set; }
+
+        }
+
+        [Fact]
+        public void ArgumentNullOnNullArgument()
+        {
+            Assert.Throws<ArgumentNullException>(() => this.TestExtractMembers<Event>(null));
+        }
+
+        [Fact]
+        public void SingleMemberAnonymousTypeImplicitMember()
+        {
+            Expression<Func<Event, object>> expression = (theEvent) => new { theEvent.Body };
+            string s = this.TestExtractMembers(expression);
```

Comment:
using var s = this.TestExtractMemebers()? I saw you use var in the StringHelperTest below, so better keep consistent style.